### PR TITLE
add particle for fruit 

### PR DESCRIPTION
**Pull request by tcm390**
1. Add crunch particle.
2. fruit size get smaller with each bite.
3. Add flash and pixel particle after consuming the fruit.
4. modify avatar emmisive color uniforms after consuming the fruit.

This is the result:

https://user-images.githubusercontent.com/60634884/163490595-70d49434-d8e1-46f9-b18d-3ada2c0d00a6.mp4

### CODE REVIEW

**Comment by alisaad673** [comment]
There are a few issues with this
1-This error appears
image

**Comment by tcm390** [comment]
Maybe I should implement the fresnel effect to make that glowing look better.

**Comment by avaer** [comment]
Is this ready for review?

**Comment by tcm390** [comment]
glowemissive
Hi @avaer, for this I just modify the `emissiveColor` of MToonMaterial uniform. I don't know whether this is the correct way to implement it. Because I found that not all of the characters using `MToonMaterial`. Even some characters do not have the emissive color. Could you please give me directions for this? Should I remove this glowing or should I just play this glowing for the character who has the emissive color.

**Comment by tcm390** [comment]
Yes, but in above video, I modify the avatar emissive color value. Now I change it to modifying the avatar color value. Because I found that not all of the characters using `MToonMaterial`. Is it okay? This is the result now:

https://user-images.githubusercontent.com/60634884/164104689-3b01d18d-ad72-419c-8d5e-9ee0a4e905ab.mp4

https://user-images.githubusercontent.com/60634884/164104710-e6a71b2c-171f-4402-9bd8-2ea810c48a97.mp4


https://user-images.githubusercontent.com/60634884/164115259-61d3ef78-3083-4e62-9b4d-4918b67be14c.mp4

**Comment by tcm390** [comment]
I would also ask @alisaad673 to help me test this again.

**Comment by tcm390** [comment]
Modify the fragmentShader `rimColor` uniform. This is the result now:

https://user-images.githubusercontent.com/60634884/164304237-d4bea742-10dc-4735-8571-49fa771ab056.mp4


https://user-images.githubusercontent.com/60634884/164304248-5c360524-d98a-409d-a685-b35c4c6b8d0c.mp4

File: fruit/fruit.js [hunk 1]
```diff
@@ -50,7 +411,51 @@ export default () => {
...
+      let removeFruitFromApp = false;
+      let circlePlay = false;
+      let materialStartTime = -1;
+      let dir = new THREE.Vector3();
+      let storeMaterial = false;
+      let materials=[];
       frameCb = (timestamp, timeDiff) => {
+
+        //#################### store avatar material ############################
+        if(localPlayer.avatar && !storeMaterial){
```
Comment (avaer) [inline fruit/fruit.js]: This looks great, but we should put this in the `app`, not the fruits.

There is already similar code for the powerup hair here: https://github.com/webaverse/app/blob/376df93433b51ef41587ee44c75e8fd8bde5f29f/character-fx.js#L179

Comment (avaer) [reply fruit/fruit.js]: Probably best to make this an action on the player (like `type: 'cure'`) and let the character fx code handle it.

**Comment by alisaad673** [comment]
looks good. Tested it with This pr:
https://github.com/webaverse/app/pull/2873#issuecomment-1110211430

**Review by avaer — APPROVED** [review]

**Pull request merged**